pyulog/core.py

```diff
@@ -537,6 +610,12 @@ def _read_file_data(self, message_name_filter_list, read_until=None):
             while True:
                 data = self._file_handle.read(3)
                 header.initialize(data)
+
+                # try recovery with sync sequence in case of unknown msg_type
+                if ((header.msg_type_str == "UNKNOWN") and self._has_sync):
```

Comment:
This changes existing behavior: newly added messages will be treated as corruption by older parsers. The current parser would just ignore those. I'm aware that we cannot have both perfectly w/o CRC. There's already existing logic to handle this.
Please use and integrate with the logic from here to check for corruption: https://github.com/PX4/pyulog/blob/master/pyulog/core.py#L603.

Reply:
I does not change the behavior when there is no sync message (which is true for existing code out there), newly added messages will go through this block ones, not find the sync bytes, and fallback to existing error handling code.

There _is_ a change in behavior for tackling new message_type messages when there exists a sync byte sequence in the file, which chooses to skip to the sync sequence rather than jump to a random number of bytes assuming that there is no corruption (the 10K size check you linked there is rather hacky). I did this because that is the only sure way of handling this situation, which is precisely what the sync sequence is there for. The unknown sync byte error is printed out, so the user can update/add handling for it.

Reply:
I could, and probably should, declare the file corrupt though if I skip bytes to the sync sequence. Is that, in essence, what you were referring to?

Reply:
> which chooses to skip to the sync sequence rather than jump to a random number of bytes assuming that there is no corruption (the 10K size check you linked there is rather hacky and would snowball)

It's not random if it's not corrupt. And while it's an approximation, it has proven to work in practice.
Your change breaks backwards compatibility: logs with newly added message types will be regarded as corrupt. This will make additions like your tagged logging message more difficult in future. ULog has ways of dealing with it (there are incompatibility flags to refuse parsing, but it will make the update process more difficult and will force all users to update). I'm not saying we cannot do that, but I like the current behavior, and I currently don't see that we have to give that up.

> The unknown sync byte error is printed out

printf is not the right approach for user reporting here. Take Flight Review for example, how would a user see this? It needs an API that can be queried. You can use printf, but guard them behind `_debug`.

> I could, and probably should, declare the file corrupt though if I skip bytes to the sync sequence. Is that, in essence, what you were referring to?

Yes, one of my points. You are currently just ignoring existing logic.